org.eclipse.languageserver/src/org/eclipse/languageserver/LanguageServiceAccessor.java

```diff
@@ -111,14 +114,34 @@ public boolean equals(Object obj) {
 		public final URI fileUri;
 		public final IFile file;
 		public final IDocument document;
-		public final LanguageClientEndpoint languageClient;
+		private final LanguageClientEndpoint languageClient;
 
 		public LSPDocumentInfo(URI fileUri, IFile file, IDocument document, LanguageClientEndpoint languageClient) {
 			this.fileUri = fileUri;
 			this.file = file;
 			this.document = document;
 			this.languageClient = languageClient;
 		}
```

Comment:
Related here: fileUri, document and language client should be check for non-nullability. Feel free to add a dependency to JDT's Nullable annotation if it helps.